gsy_framework/data_classes.py

```diff
@@ -291,10 +349,9 @@ def accumulated_grid_fees(self):
         """Return the accumulated grid fees alongside the path of the bid."""
         return self.original_price - self.price
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: "Bid") -> bool:
         return (self.id == other.id and
                 self.buyer == other.buyer and
-                self.buyer_origin_id == other.buyer_origin_id and
```

Comment:
It is actually one side-fix here :D The equality of the orders should not just include the origin of the order, since two different orders might share the same origin (e.g. one load posting 2 orders with different amounts of energy). Admittedly the id would protect us in these cases (essentially the orders would have different ids) nevertheless it does not hurt to add the full traderDetails object as an extra check.